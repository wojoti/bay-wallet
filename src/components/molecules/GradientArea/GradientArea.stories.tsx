import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Text} from 'react-native';
import {green, lime} from 'theme/colors';

import GradientArea from './GradientArea';

export default {
  title: 'Molecules/GradientArea',
  component: GradientArea,
} as ComponentMeta<typeof GradientArea>;

const Template: ComponentStory<typeof GradientArea> = args => (
  <GradientArea {...args}>
    <Text>Added required component</Text>
  </GradientArea>
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  colors: [green, lime],
  start: {
    x: 0,
    y: 0,
  },
  end: {
    x: 1,
    y: 1,
  },
};
