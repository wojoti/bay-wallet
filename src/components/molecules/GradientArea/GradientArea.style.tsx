import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  gradient: {
    flex: 1,
    borderRadius: 14,
  },
  wrapper: {
    borderRadius: 14,
  },
});
export default style;
