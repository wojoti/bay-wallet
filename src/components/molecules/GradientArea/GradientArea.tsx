import {Container} from 'components/atoms';
import {PropsWithChildren} from 'react';
import LinearGradient from 'react-native-linear-gradient';
import style from './GradientArea.style';

export type GradientAreaProps = PropsWithChildren<{
  colors: (string | number)[];
  start?: {
    x: number;
    y: number;
  };
  end?: {
    x: number;
    y: number;
  };
  testId?: string;
}>;

const GradientArea = ({
  testId,
  colors,
  start,
  end,
  children,
}: GradientAreaProps) => {
  return (
    <Container flex={1} customStyle={style.wrapper} testId={testId}>
      <LinearGradient
        start={start}
        end={end}
        style={style.gradient}
        colors={colors}
        testID="gradientarea-gradient">
        {children}
      </LinearGradient>
    </Container>
  );
};

export default GradientArea;
