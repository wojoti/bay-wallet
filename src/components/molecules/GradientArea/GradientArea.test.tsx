import '@testing-library/jest-native/extend-expect';
import {render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import GradientArea, {GradientAreaProps} from './GradientArea';

const props: GradientAreaProps = {
  testId: 'test-gradientarea-id',
  colors: ['green', 'lime'],
  start: {
    x: 0,
    y: 0,
  },
  end: {
    x: 1,
    y: 1,
  },
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<GradientArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render gradientarea', () => {
  render(<GradientArea {...props} />);
  const gradientareaElement = screen.getByTestId('test-gradientarea-id');
  expect(gradientareaElement).toBeOnTheScreen();
});

test('should render gradientarea - gradient', () => {
  render(<GradientArea {...props} />);
  const gradientareaElement = screen.getByTestId('test-gradientarea-id');
  expect(gradientareaElement).toBeOnTheScreen();
  const linearGradient = within(gradientareaElement).getByTestId(
    'gradientarea-gradient',
  );
  expect(linearGradient).toBeOnTheScreen();
});
