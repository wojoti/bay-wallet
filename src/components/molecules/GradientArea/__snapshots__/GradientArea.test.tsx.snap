// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "borderRadius": 14,
      },
    ]
  }
  testID="test-gradientarea-id"
>
  <BVLinearGradient
    colors={
      [
        4278222848,
        4278255360,
      ]
    }
    endPoint={
      {
        "x": 1,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "borderRadius": 14,
        "flex": 1,
      }
    }
    testID="gradientarea-gradient"
  />
</View>
`;
